src/v2/System/useFeatureFlag.tsx

```diff
@@ -60,15 +60,14 @@ export function useTrackVariantView({
   contextOwnerSlug,
   shouldTrackExperiment = true,
 }: VariantTrackingProperties) {
-  const { trackEvent } = useTracking()
-
   useEffect(() => {
-    if (typeof window !== "undefined") {
+    if (
+      typeof window !== "undefined" &&
+      typeof window.analytics !== "undefined"
+    ) {
       const trackFeatureView = shouldTrack(experimentName, variantName)
-
       if (trackFeatureView && shouldTrackExperiment) {
-        trackEvent({
-          action: ActionType.experimentViewed,
+        window.analytics.track(ActionType.experimentViewed, {
```

Comment:
After a quick QA round we found out that `trackEvent` wasn't behaving as expected and was almost never called. Replacing that with `window.analytics.track` fixed the issue and now analytics are being tracked 100% of the time.

Let me know what you think about this

Reply:
Would very much be interested in knowing why?

Reply:
cc @damassi

Reply:
I would be really interested in knowing that as well not 100% sure the why but it wasn't called when expected, happy to pair on this if you have any other suggestion.

Reply:
When pulling the trackEvent on a place like a button and we pressed it it was tracking as expected, but inside the `useEffect` there it is not working and I really don't know the why

Reply:
I'd at least leave a comment explaining this. This is the kind of thing I'd immediately refactor to use trackEvent if I came across it.

Reply:
done! 👍

Reply:
I've [ticketed out the follow-up](https://artsyproduct.atlassian.net/browse/PLATFORM-4059) to take another look at the analytics approach. Do we want to move forward with this approach to continue testing, while we continue to look at the analytics question separately?